### PR TITLE
[arrow] Update to 12.0.1

### PR DESCRIPTION
**Pull request by raulcd**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/arrow/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "arrow",
-  "version": "12.0.0",
+  "version": "12.0.1",
   "port-version": 2,
```
Comment (Adela0814) [inline ports/arrow/vcpkg.json:4]: Please remove the port-version.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/arrow/vcpkg.json (hunk 1 above)
Comment (raulcd) [reply ports/arrow/vcpkg.json]: Thanks!

**Comment by Adela0814** [comment]
Tested all feature on the following triplets:

- x64-windows
- x86-windows
- x64-windows-static

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**